extensions/ql-vscode/src/remote-queries/sarif-processing.ts

```diff
@@ -19,77 +20,63 @@ export function extractAnalysisAlerts(
   alerts: AnalysisAlert[],
   errors: string[]
 } {
-  if (!sarifLog) {
-    return { alerts: [], errors: ['No SARIF log was found'] };
-  }
-
-  if (!sarifLog.runs) {
-    return { alerts: [], errors: ['No runs found in the SARIF file'] };
-  }
-
-  const errors: string[] = [];
   const alerts: AnalysisAlert[] = [];
+  const errors: string[] = [];
 
-  for (const run of sarifLog.runs) {
-    if (!run.results) {
-      errors.push('No results found in the SARIF run');
-      continue;
-    }
-
-    for (const result of run.results) {
-      if (!result.message?.text) {
-        errors.push('No message found in the SARIF result');
-        continue;
-      }
-
-      const { message, errors: messageErrors } = extractMessage(result.message?.text, result);
-      if (messageErrors.length > 0) {
-        errors.push(...messageErrors);
-        continue;
-      }
-
-      const rule = tryGetRule(run, result);
-      const severity = tryGetSeverity(run, result, rule) || defaultSeverity;
-
-      const { codeFlows, errors: codeFlowsErrors } = extractCodeFlows(result);
-      if (codeFlowsErrors.length > 0) {
-        errors.push(...codeFlowsErrors);
-        continue;
-      }
-
-      if (!result.locations) {
-        errors.push('No locations found in the SARIF result');
-        continue;
-      }
-
-      const shortDescription = getShortDescription(rule, message!);
-
-      for (const location of result.locations) {
-        const { processedLocation, errors: locationErrors } = extractLocation(location);
-
-        if (locationErrors.length > 0) {
-          errors.push(...locationErrors);
+  try {
+    for (const run of sarifLog.runs!) {
+      for (const result of run.results!) {
```

Comment:
To be safe, maybe something like this:

```suggestion
    for (const run of sarifLog.runs ?? []) {
      for (const result of run.results ?? []) {
```

It's _probably_ fine, but not sure what would happen if a run legitimately has no results.

Reply:
Based on what I've seen in the code base etc. I don't think it's possible, but I guess it doesn't hurt adding the fallbacks.